crowbar_framework/app/controllers/nodes_controller.rb

```diff
@@ -307,19 +307,42 @@ def hit
     action = params[:req]
     name = params[:name] || params[:id]
     machine = NodeObject.find_node_by_name name
-    if machine.nil?
-      render text: "Could not find node '#{name}'", status: 404 and return
-    else
-      case action
-      when "reinstall", "reset", "update", "delete"
-        machine.set_state(action)
-      when "reboot", "shutdown", "poweron", "powercycle", "poweroff", "identify", "allocate"
-        machine.send(action)
-      else
-        render text: "Invalid hit request '#{action}'", status: 500 and return
+    respond_to do |format|
+      format.json do
+        if machine.nil?
+          render json: { error: "Could not find node '#{name}'" }, status: 404 && return
+        else
+          case action
+          when "reinstall", "reset", "update", "delete"
+            machine.set_state(action)
+          when "reboot", "shutdown", "poweron", "powercycle", "poweroff", "identify", "allocate"
+            machine.send(action)
+          else
+            render json: { error: "Invalid hit request '#{action}'" }, status: 500 && return
+          end
+            head :ok
+        end
+      end
+
+      format.html do
+        if machine.nil?
+          flash[:alert] = "Could not find node '#{name}'"
+          redirect_to nodes_path, status: 404
```

Comment:
I think that should be `dashboard_url`. Hint: Always use the `_url` helpers for redirects.